authenticatorbridge/src/main/java/com/bitwarden/authenticatorbridge/manager/AuthenticatorBridgeManagerImpl.kt

```diff
@@ -0,0 +1,195 @@
+            }
+
+            if (symmetricKeyStorageProvider.symmetricKey == null) {
+                // This means bridgeService returned a null key, which means we can make
+                // no valid operations. We should disconnect form the service and expose to the
+                // calling application that authenticator sync is not enabled.
+                mutableSharedAccountsStateFlow.value = AccountSyncState.SyncNotEnabled
+                unBindService()
+            }
+
+            // Register callback:
+            bridgeService.safeCall { registerBridgeServiceCallback(authenticatorBridgeCallback) }
+
+            // Sync data:
+            bridgeService.safeCall { syncAccounts() }
+        }
+
+        override fun onServiceDisconnected(name: ComponentName) {
+            bridgeService = AuthenticatorBridgeServiceSafetyWrapper(service = null)
+        }
+    }
+
+    override val accountSyncStateFlow: StateFlow<AccountSyncState> =
+        mutableSharedAccountsStateFlow.asStateFlow()
+
+    init {
+        // Listen for lifecycle events
+        processLifecycleOwner.lifecycle.addObserver(
+            object : DefaultLifecycleObserver {
+                override fun onStart(owner: LifecycleOwner) {
+                    bindService()
+                }
+
+                override fun onStop(owner: LifecycleOwner) {
+                    unBindService()
+                }
+            },
+        )
+    }
+
+    private fun bindService() {
+        val intent = Intent().apply {
+            component = ComponentName(
+                connectionType.toPackageName(),
+                AUTHENTICATOR_BRIDGE_SERVICE_CLASS,
+            )
+        }
+
+        isBound = try {
+            applicationContext.bindService(
+                intent,
+                bridgeServiceConnection,
+                Context.BIND_AUTO_CREATE,
+            )
+        } catch (e: SecurityException) {
+            false
+        }
+
+        if (!isBound) {
+            mutableSharedAccountsStateFlow.value = AccountSyncState.Error
+        }
+    }
+
+    private fun onAccountsSync(data: EncryptedSharedAccountData) {
+        // Received account sync update. Decrypt with local symmetric key and update StateFlow:
+        val decryptedAccounts = symmetricKeyStorageProvider.symmetricKey?.let {
+            val decryptedAccounts = data.decrypt(it).getOrNull()?.accounts ?: return@let null
+            AccountSyncState.Success(decryptedAccounts)
+        }
+        mutableSharedAccountsStateFlow.value = decryptedAccounts
+            ?.let { AccountSyncState.Success(it.accounts) }
+            ?: AccountSyncState.Error
+    }
+
+    private fun unBindService() {
+        if (!isBound) return
+        isBound = false
+        bridgeService.safeCall { unregisterBridgeServiceCallback(authenticatorBridgeCallback) }
+        bridgeService = AuthenticatorBridgeServiceSafetyWrapper(service = null)
```

Comment:
I'd really prefer to just null this whole thing out and ditch the wrapper class. (See comment below.)